src/Sylius/Bundle/AdminBundle/Controller/ImpersonateUserController.php

```diff
@@ -26,32 +26,19 @@
 
 final class ImpersonateUserController
 {
-    private UserImpersonatorInterface $impersonator;
-
-    private AuthorizationCheckerInterface $authorizationChecker;
-
-    private UserProviderInterface $userProvider;
-
     private ?RouterInterface $router;
 
-    private string $authorizationRole;
-
     public function __construct(
-        UserImpersonatorInterface $impersonator,
-        AuthorizationCheckerInterface $authorizationChecker,
-        UserProviderInterface $userProvider,
+        private UserImpersonatorInterface $impersonator,
+        private AuthorizationCheckerInterface $authorizationChecker,
+        private UserProviderInterface $userProvider,
         ?RouterInterface $router,
```

Comment:
```suggestion
        private ?RouterInterface $router,
```

Reply:
But to be honest, we should rework this argument, to be able to replace logic with the new one

Reply:
Interesting idea, we can elaborate over it in the separate PR 🖖